extensions/resteasy-reactive/rest-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/ResteasyReactiveJacksonProcessor.java

```diff
@@ -600,24 +622,34 @@ private static boolean anyParentClassHasSecureFields(IndexView indexView, ClassI
         if (!currentClassInfo.superName().equals(ResteasyReactiveDotNames.OBJECT)) {
             final ClassInfo parentClassInfo = indexView.getClassByName(currentClassInfo.superName());
             return parentClassInfo != null
-                    && hasSecureFields(indexView, parentClassInfo, typeToHasSecureField, needToDeleteCache);
+                    && hasSecureFields(indexView, parentClassInfo, typeToHasSecureField, needToDeleteCache, null);
         }
         return false;
     }
 
     private static boolean anySubclassHasSecureFields(IndexView indexView, ClassInfo currentClassInfo,
             Map<String, Boolean> typeToHasSecureField, AtomicBoolean needToDeleteCache) {
         return indexView.getAllKnownSubclasses(currentClassInfo.name()).stream()
-                .anyMatch(subclass -> hasSecureFields(indexView, subclass, typeToHasSecureField, needToDeleteCache));
+                .anyMatch(subclass -> hasSecureFields(indexView, subclass, typeToHasSecureField, needToDeleteCache, null));
     }
 
     private static boolean anyFieldHasSecureFields(IndexView indexView, ClassInfo currentClassInfo,
-            Map<String, Boolean> typeToHasSecureField, AtomicBoolean needToDeleteCache) {
+            Map<String, Boolean> typeToHasSecureField, AtomicBoolean needToDeleteCache,
+            Map<String, Type> typeParamIdentifierToParameterizedType) {
         return currentClassInfo
                 .fields()
                 .stream()
                 .filter(fieldInfo -> !fieldInfo.hasAnnotation(JSON_IGNORE))
                 .map(FieldInfo::type)
+                .map(fieldType -> {
```

Comment:
We have better type mapping functions already in place that do better than just handling simple cases like the field type being a type variable. Probably this could be improved.

Reply:
I was thinking of the logic we already do to resolve type parameters in resteasy reactive. I think via `JandexUtil` and `TypeMapper`, but I can only find usages of them to obtain method parameter or return type signatures or the list of type parameters. I can't find anywhere where we resolve an entire `Type` with type parameters substituted. Probably I dreamed this :(

Reply:
np, if you run into this or @geoand knows, please let me know. it's not on the top of my list tbh, so no hurry

Reply:
> I can't find anywhere where we resolve an entire Type with type parameters substituted. Probably I dreamed this

AFAIR, the cases we've it for are method parameters and return types